### PR TITLE
Added Docs for NerdGraph API in New Relic connector

### PR DESCRIPTION
**Pull request by SushrutHarness**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [X] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [X] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 1]
```diff
@@ -90,11 +90,24 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
...
 For secrets and other sensitive settings, select or create a new [Text Secret](/docs/platform/secrets/add-use-text-secrets). Enter the API key needed to connect with the server.
 
-For steps on generating the New Relic API key, follow this doc from New Relic: [Insights query API](https://docs.newrelic.com/docs/apis/insights-apis/query-insights-event-data-api/).
+If you are having trouble connecting to the API, follow the docs linked below depending on if you're using the NerdGraph API or the Insight Query API.
 
-If you have trouble finding step on generating the **Insights query key**, look for the API key types help in the New Relic help panel:
+<details>
+<summary> NerdGraph API </summary>
+   
+   - For steps on connecting to the NerdGraph API, follow these docs from New Relic: [NerdGraph API](https://docs.newrelic.com/docs/apis/nerdgraph/get-started/introduction-new-relic-nerdgraph/)
```
Comment (brian-f-harness) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:98]:
```suggestion
   - For steps on connecting to the NerdGraph API, go to [NerdGraph API](https://docs.newrelic.com/docs/apis/nerdgraph/get-started/introduction-new-relic-nerdgraph/) in the New Relic documentation.
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 2]
```diff
@@ -90,11 +90,24 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
...
+   - You gain access to this API by using the `user key` for a specific account.
+
+</details>
+
+<details>
+<summary> Insights Query API (DEPRECATED) </summary>
+   - For steps on generating the New Relic API key, follow this doc from New Relic: [Insights query API](https://docs.newrelic.com/docs/apis/insights-apis/query-insights-event-data-api/).
```
Comment (brian-f-harness) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:105]:
```suggestion
   - For steps on generating the New Relic API key, go to [Insights query API](https://docs.newrelic.com/docs/apis/insights-apis/query-insights-event-data-api/) in the New Relic documentation.
```

**Review by brian-f-harness — APPROVED** [review]

**Review by brian-f-harness — DISMISSED** [review]

File: docs/continuous-delivery/cd-integrations.md [hunk 3]
```diff
@@ -235,4 +235,8 @@ To enable a feature flag in your Harness account, contact [Harness Support](mail
         <td>CDS_ECS_BG_VALIDATION</td>
         <td>If green services exist in your Blue Green deployment, you can configure Harness to update those services instead of deleting them and then re-creating them with a new manifest and artifact.<br/>For more information, go to <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#update-green-services">Update green services</a> and <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#ecs-blue-green-service-validation">ECS blue/green service validations</a>.<br/><b>This feature is in Beta, and it requires Harness Delegate version 23.11.81820.</b> </td>
     </tr>
+     <tr>
+        <td>CV_NEWRELIC_NEW_API</td>
+        <td>Enable this feature if you want to use the NerdGraph API for the NewRelic Health Source. <b>This feature is in Beta.</b></td>
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-integrations.md:240]:
```suggestion
        <td>Enable this feature if you want to use the NerdGraph API for the NewRelic Health Source. <b>This is a Beta feature.</b></td>
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 4]
```diff
@@ -80,7 +80,9 @@ You cannot use a Grafana URL.
 6. In the **New Relic** **URL** field, select the URL of your New Relic Insights API. 
 
 :::info note
-New Relic supports multiple APIs. The Harness Connector leverages the New Relic Insights API, which supports NRQL. See Step 8 for creating an Insights API Key.
+New Relic supports multiple APIs. The Harness Connector leverages the NerdGraph API or the deprecated New Relic Insights API, which supports NRQL. See Step 8 for more details on connecting to either API.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:83]:
```suggestion
New Relic supports multiple APIs. Harness connector leverages the NerdGraph API or the deprecated New Relic Insights API that supports NRQL. See Step 8 for more details on connecting to either APIs.
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 5]
```diff
@@ -80,7 +80,9 @@ You cannot use a Grafana URL.
...
+
+Currently, the NerdGraph API is behind the feature flag `CV_NEWRELIC_NEW_API`. Please contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:85]:
```suggestion
Currently, the NerdGraph API is behind the feature flag `CV_NEWRELIC_NEW_API`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```

Comment (roshnisarangadharan) [reply docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:85]: We don't say please or any request kind of language in technical docs.

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 6]
```diff
@@ -90,11 +92,28 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
 
 For secrets and other sensitive settings, select or create a new [Text Secret](/docs/platform/secrets/add-use-text-secrets). Enter the API key needed to connect with the server.
 
-For steps on generating the New Relic API key, follow this doc from New Relic: [Insights query API](https://docs.newrelic.com/docs/apis/insights-apis/query-insights-event-data-api/).
+If you are having trouble connecting to the API, follow the docs linked below depending on if you're using the NerdGraph API or the Insight Query API.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:95]:
```suggestion
If you are having trouble connecting to the API, follow the docs linked below depending on whether you're using the NerdGraph API or the Insight Query API.
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 7]
```diff
@@ -90,11 +92,28 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
...
+
+<details>
+<summary> NerdGraph API </summary>
+   
+   :::note
+   Currently, this feature is behind the feature flag `CV_NEWRELIC_NEW_API`. Please contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:101]:
```suggestion
   Currently, this feature is behind the feature flag `CV_NEWRELIC_NEW_API`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 8]
```diff
@@ -90,11 +92,28 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
...
+   :::
+
+   - For steps on connecting to the NerdGraph API, go to [NerdGraph API](https://docs.newrelic.com/docs/apis/nerdgraph/get-started/introduction-new-relic-nerdgraph/) in the New Relic documentation.
+   - You gain access to this API by using the `user key` for a specific account.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:105]:
```suggestion
   - To gain access to this API, use the `user key` for a specific account.
```

File: docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md [hunk 9]
```diff
@@ -90,11 +92,28 @@ New Relic supports multiple APIs. The Harness Connector leverages the New Relic
...
+
+</details>
+
+<details>
+<summary> Insights Query API (DEPRECATED) </summary>
+   - For steps on generating the New Relic API key, go to [Insights query API](https://docs.newrelic.com/docs/apis/insights-apis/query-insights-event-data-api/) in the New Relic documentation.
 
-If you have trouble finding step on generating the **Insights query key**, look for the API key types help in the New Relic help panel:
+   - If you have trouble finding step on generating the **Insights query key**, look for the API key types help in the New Relic help panel:
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/monitoring-and-logging-systems/connect-to-monitoring-and-logging-systems.md:113]:
```suggestion
   - If you have trouble finding the step on generating the **Insights query key**, look for the API key types help in the New Relic help panel:
```

**Review by roshnisarangadharan — APPROVED** [review]

**Pull request merged**